hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/FindSourceGreedy.java

```diff
@@ -138,6 +138,11 @@ public void removeCandidateSourceDataNode(DatanodeDetails dui) {
     potentialSources.removeIf(a -> a.getDatanodeDetails().equals(dui));
   }
 
+  @Override
+  public void addBackSourceDataNode(DatanodeDetails dui) {
+    potentialSources.add(nodeManager.getUsageInfo(dui));
+  }
```

Comment:
A caller could use this method to a Datanode that wasn't previously removed from the priority queue. This could result in multiple instances of the same Datanode in the queue. Let's add a javadoc to the interface and explicitly call out that anyone using this method should ensure the Datanode was previously removed or that it isn't present in the queue.